dialer.go

```diff
@@ -565,3 +597,43 @@ func (d *Dialer) connectionInfoCache(
 
 	return c, nil
 }
+
+func (d *Dialer) queryDNS(ctx context.Context, domainName string) (instance.ConnName, error) {
```

Comment:
Since this is a non-trivial method, let's add a godoc (even though it's private) for future readers.

Reply:
Method comment added.